src/Adapter/AdapterAbstract.php

```diff
@@ -52,12 +54,8 @@
      */
     public function ok($output)
     {
-        $lines = explode("\n", $output);
-        foreach ($lines as $line) {
-            if (preg_match("%not ok \\d+ - %", $line)
-            && !preg_match("%# TODO%", $line)) {
-                return false;
-            }
+        if (preg_match("%##teamcity\[testFailed%", $output)) {
```

Comment:
but isn't it what is done with the if statement? Too me both look strictly equivalent